src/async_op.c

```diff
@@ -35,30 +41,69 @@ static void async_op_dispose(ASYNC_OP* async_op)
 #define IS_POWER_OF_2(x) \
     (((x) & ((x) - 1)) == 0)
 
+/*tiny function that returns the next multiple of "align" starting from "start" (included). Does 1 mod and 1 multiplication*/
+static void* get_next_aligned_address(void* start, uint32_t align)
+{
+    void* result;
+
+    /*compute the misalignment of the address*/
+    uint32_t m = (uintptr_t)start % align;
+    
+    /*if it is aligned then m!=0 evaluates to 0. Anything multiplied by 0 is 0. Otherwise m!=0 evaluates to "1" and then the next address is start - m + align*/
+    result = (void*)((uintptr_t)start + (m!=0) * (align-m));

```

Comment:
>(void*) [](http://example.com/codeflow?start=13&length=7)

What happens if this cast is removed?

Reply:
`warning C4047: '=': 'void *' differs in levels of indirection from 'uintptr_t'` happens